### ISSUE TITLE
'Failed to execute child' when clicking a link the name of which contains a space

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/nemo#2568]
```
 * Nemo version (nemo --version) 4.6.5
 * Is issue with desktop or windowed nemo? Both
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 20
 * Graphics hardware *and* driver used: not relevant
 * 32 or 64 bit: 64
 ```

**Issue**

![image](https://user-images.githubusercontent.com/47056543/100040779-d6e25680-2dff-11eb-8d08-a2a707154409.png)

**Steps to reproduce**

1. Have a executable file - say, a Bash script (set as executable) - that contains a space in its name. Use Nemo to make a link to it. (The link can be left where it is created.)
2. Run it via Nemo, with Nemo set to ask what to do when running executables - choose 'run in terminal' when asked.

**Expected behaviour**

The executable should run.

**Other information**

I have the problem using both Gnome terminal and xfce4-terminal.
File manager PcManFm does not have the problem - it runs the executable.

**Comment by icarter09** [comment]
@LinuxOnTheDesktop is this a duplicate of #2499?

**Comment by LinuxOnTheDesktop** [comment]
It is. Sorry - I had not seen that report.

One comment on that report says, 'It's your bash scripts that are broken not nemo.' I do not understand. For, my affected scripts do not have trouble running some other script. Rather the problem is running the script itself. Still, on that same page, clem says: no, this is a bug.

I note also that that page seems to recommend running `gsettings set org.cinnamon.desktop.default-applications.terminal exec-arg -- `. Should I try that? Thanks. Still: closing this issue, because it is a duplicate.

**Comment by icarter09** [comment]
@LinuxOnTheDesktop yes please give that setting a try and see if that works for you. Thanks.

**Comment by LinuxOnTheDesktop** [comment]
@icarter09 

Thanks. That gsettings command makes the operation work when Gnome-terminal is set as the default terminal but _not_ when xfce4-terminal is the default. If the latter is the default, then clicking a link to such a script, and then choosing 'run in terminal' produces - by way of visible output - nothing at all.

**Comment by icarter09** [comment]
@LinuxOnTheDesktop thanks for confirming that. The issue is not the spacing, but lies in the fact of not setting the proper value for the given terminal. I have a PR to address the issue, but like you said it's only with gnome-terminal. Feel free to create a new issue xfce4-terminal and I'll look into putting together a fix it. Thanks.

**Comment by LinuxOnTheDesktop** [comment]
Done (#2571). Should I revert the gsetting setting? If so, how?

**Comment by icarter09** [comment]
Two ways are 1) via UI, using dconf-editor and setting it that way (org/cinnamon/desktop/default-applications/terminal/exec-arg) or 2) run the same cmd and change the `--` to whatever option you want.

**Comment by LinuxOnTheDesktop** [comment]
1) So you are saying I should revert it?
2) You do not seem to be telling me just what to revert it to. '[W]hatever option you want' - what I want is the default. What is that, please?

**Comment by icarter09** [comment]
1. Yes revert it since it isn't working.
2. I updated the comment to include what options you should revert it to.

**Comment by LinuxOnTheDesktop** [comment]
@icarter09 

Thanks, but you do not seem to have actually done 2. Also, the dconf path you supplied seems wrong. The correct path seems to be `org.cinnamon.desktop.applications.terminal.exec-arg`. And/but I used the dconf editor on another machine to see what the default is. It said the default was `-x` and the current value - on, to repeat, a machine I had not run the gsettings command on - was `-e`. So I set the current value, on the machine I'd run the gsettings command on, to `-e`. So that's fine.
